### DISCUSSION TITLE
Upgrade of RetroArch keep removing files

### DISCUSSION THREAD

**Discussion by markand** [Homebrew/discussions#2286] [Casks]
Hi,

It looks like upgrades using `brew upgrade --cask` always zap files from RetroArch (assets, core, databases and everything else) because they are store in ~/Library.

I wonder why brew remove those files while upgrading? I mean, when removing using `--zap` option I can understand, but each time I upgrade RetroArch I need to download cores manually again.

Is there a thing to prevent that? Should we remove this removal of files on upgrades?

**Comment by SMillerDev** [comment]
As far as I can tell [the cask](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/retroarch.rb) doesn't define anything to be deleted. Do you have an example of the files that are deleted on updates?

**Reply by markand** [reply]
Hmm, no I get it...

By default RetroArch download files inside its own .app directory. e.g.:

- /Applications/RetroArch.app/Contents/Resources/cores
- /Applications/RetroArch.app/Contents/Resources/database
- etc...

On Linux distributions we usually patch the default retroarch.cfg to set different paths for those files (because on Linux we need to set a user directory so it can use the download updater to download file as regular users). It's common idiom to do that on per Linux distribution basis.

Would it be okay if we do the same for homebrew? Patching the retroarch.cfg so that files are downloaded inside the user ~/Library folder (there is already ~/Library/Application Support/RetroArch/{cores,database} by default).

**Reply by SMillerDev** [reply]
It seems like it would be much better if that was the default so a pull request to RetroArch to change this default would be better IMO.
